harness-delegate-ng/templates/ccm/cost-access.yaml

```diff
@@ -0,0 +1,76 @@
+{{- if .Values.ccm.visibility }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ .Values.ccm.role_name }}
+  labels:
+    {{- include "harness-delegate-ng.labels" . | nindent 4 }}
+rules:
```

Comment:
The list of permissions is quite extensive, are we certain that we need all of them (and with all the verbs)?
What if delegate is already configured to be cluster admin, do we even need this additional cluster role at all? If not can we skip it altogether if delegate is cluster admin?

Reply:
The list of permissions comes directly from the CCM setup guide.

This feature is exactly for users who are **not** making their delegate cluster admin, and CLUSTER_VIEWER also dose not have the correct permissions.

These are primary CCM-only customers so they are not using delegates for deployments so they need the least access possible.

Reply:
role is now skipped if `ccm.visibility=true` and `k8sPermissionsType=CLUSTER_ADMIN`